ui/app/routes/vault/cluster.js

```diff
@@ -67,39 +68,27 @@ export default Route.extend(ModelBoundaryRoute, ClusterRoute, {
 
   model(params) {
     const id = this.getClusterId(params);
-
     return this.get('store').findRecord('cluster', id);
   },
 
-  stopPoll: on('deactivate', function() {
-    cancel(this.get('timer'));
-  }),
-
-  poll() {
+  poll: task(function*() {
     // when testing, the polling loop causes promises to never settle so acceptance tests hang
     // to get around that, we just disable the poll in tests
-    return Ember.testing
-      ? null
-      : later(() => {
-          this.controller
-            .get('model')
-            .reload()
-            .then(
-              () => {
-                this.set('timer', this.poll());
-                return this.transitionToTargetRoute();
-              },
-              () => {
-                this.set('timer', this.poll());
-              }
-            );
-        }, POLL_INTERVAL_MS);
-  },
+    do {
+      yield timeout(POLL_INTERVAL_MS);
+      try {
+        yield this.controller.model.reload();
```

Comment:
We're using `this.controller.model` here, but triggering this task from `afterModel`. While there's a timeout immediately before this, so `setupController` is almost definitely going to finish, it's possible that in future you might add a blocking promise as the result of `afterModel` which would cause a race.

Instead, you might want to trigger the task from `setupController`, or close over the `model`?

Reply:
Oh yeah good call - seems like that may have been the issue before 🤔 - will move this to `setupController`